analyses/transcriptomic-dimension-reduction/01-transcriptomic-analysis-prep.Rmd

```diff
@@ -0,0 +1,316 @@
+```{r}
+# magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+perform_dimension_reduction <- function(transposed_expression_matrix, method, seed = 2019) {
+  # Given a data.frame that containes the transposed expression matrix and the
+  # name of a dimension reduction method, perform the dimension reduction 
+  # technique on the transposed matrix
+  # Args:
+  #   transposed_expression_matrix: transposed `RSEM` or `Kallisto` expression
+  #                                 matrix
+  #   method: dimension reduction method to be performed
+  #   seed: seed set to a default of 2019
+  #
+  # Returns:
+  #   dimension_reduction_df: data.frame containing the resulting scores of the
+  #                           dimension reduction technique, with a column that
+  #                           includes sample identifiers
+  
+  # Set seed for reproducibility
+  set.seed(seed)
+  
+  # Save rownames as a vector
+  ID <- rownames(transposed_expression_matrix)
+  # Perform dimension reduction 
+  if (method == "PCA" ){
+  prcomp_results <- prcomp(transposed_expression_matrix)
+  dimension_reduction_df <- data.frame(prcomp_results$x[, 1:2])
+  } else if (method == "t-SNE" ) {
+  tsne_results <- Rtsne::Rtsne(transposed_expression_matrix, perplexity = 15)
+  dimension_reduction_df <- data.frame(tsne_results$Y)
+  } else if (method == "UMAP" ) {
+  umap_results <- umap::umap(transposed_expression_matrix)
+  dimension_reduction_df <- data.frame(umap_results$layout)
+  } else {
+    print ("Error")
+  }
+  
+  # To set the tsv filename 
+  matrix_name <- as.name(transposed_expression_matrix)
+  # Assign the relevant rownames(Kids_First_Biospecimen_ID) to a column
+  dimension_reduction_df$Kids_First_Biospecimen_ID <- ID
+  # Write the resulting data.frame to a tsv file
+  readr::write_tsv(dimension_reduction_df, 
+                   file.path(results_dir, file = paste(matrix_name, method, "scores.tsv", sep = "_")))
+  
+  return(dimension_reduction_df)
+}
+align_metadata <- function(dimension_reduction_df, metadata_df, filename) {
+  # Given a data.frame that contains scores from a dimensionality reduction
+  # technique, align the metadata to the data.frame in preparation for plotting
+  #
+  # Note: The rownames provided in the id argument are synonymous with the
+  # metadata's `Kids_First_Biospecimen_ID`
+  #
+  # Args:
+  #   dimension_reduction_df: Name of the data.frame containing dimension reduction scores
+  #   metadata_df : Name of the data.frame containing the relevant metadata
+  #
+  # Returns:
+  #   aligned_scores_df: A data.frame containing the dimension reduction scores along with
+  #         the variables from `metadata_df`
+  
+  # Join the dimension reductions scores data.frame and the metadata data.frame 
+  aligned_scores_df <- dimension_reduction_df %>%
+    dplyr::inner_join(metadata_df, by = c("Kids_First_Biospecimen_ID")) 
+  
+  # Write the resulting metadata aligned data.frame to a RDS file
+  readr::write_rds(aligned_scores_df, 
```

Comment:
I would expect this to be a TSV file so that folks that are not R users could plot, say, PC1 vs. a variable of their interest.